Runtime/Tracking/Collision/CollisionIgnorer.cs

```diff
@@ -0,0 +1,213 @@
+            Targets.ElementRemoved.RemoveListener(RemoveTarget);
+        }
+
+        /// <summary>
+        /// Adds a new source to ignore colliding with all of the <see cref="Targets"/> colliders.
+        /// </summary>
+        /// <param name="source">The source to ignore collisions from.</param>
+        protected virtual void AddSource(GameObject source)
+        {
+            ToggleCollisions(source, Sources, Targets, true);
+        }
+
+        /// <summary>
+        /// Removes an existing source from ignoring collisions with all of the <see cref="Targets"/> colliders.
+        /// </summary>
+        /// <param name="source">The source to restore collisions with.</param>
+        protected virtual void RemoveSource(GameObject source)
+        {
+            ToggleCollisions(source, Sources, Targets, false);
+        }
+
+        /// <summary>
+        /// Adds a new target to ignore colliding on from of the <see cref="sources"/> colliders.
+        /// </summary>
+        /// <param name="target">The target to ignore collisions on.</param>
+        protected virtual void AddTarget(GameObject target)
+        {
+            ToggleCollisions(target, Targets, Sources, true);
+        }
+
+        /// <summary>
+        /// Removes an existing target from having collisions ignored with all of the <see cref="sources"/> colliders.
+        /// </summary>
+        /// <param name="target">The target to restore collisions on.</param>
+        public virtual void RemoveTarget(GameObject target)
+        {
+            ToggleCollisions(target, Targets, Sources, false);
+        }
+
+        /// <summary>
+        /// Sets the collision state between <see cref="Sources"/> and <see cref="Targets"/>.
+        /// </summary>
+        /// <param name="state">Whether to ignore collisions or not.</param>
+        protected virtual void ToggleCollisions(bool state)
+        {
+            foreach (GameObject source in Sources.SubscribableElements)
+            {
+                ToggleCollisions(source, Sources, Targets, state);
+            }
+        }
+
+        /// <summary>
+        /// Sets the collision state between the source and targets.
+        /// </summary>
+        /// <param name="source">The source to set the collision state on.</param>
+        /// <param name="targets">A collection of targets to set the collision state on.</param>
+        /// <param name="state">Whether to ignore collisions or not.</param>
+        protected virtual void ToggleCollisions(GameObject source, GameObjectObservableList sources, GameObjectObservableList targets, bool state)
+        {
+            if (source == null || (!state && sources.Contains(source)))
+            {
+                return;
+            }
+
+            List<Collider> sourceColliders = new List<Collider>();
+            List<Collider> targetColliders = new List<Collider>();
+
+            sourceColliders.AddRange(source.GetComponentsInChildren<Collider>());
```

Comment:
There's an overload of the `GetComponentsInChildren` method that *takes* a list. Use that for both these calls together with two lists you reuse by making it a `protected` field instead. Note that you do **not** have to call `Clear` on those lists - `GetComponentsInChildren` does that for you already.

This means we reduce the amount of GC allocations since the list will not change in capacity (not size) all the time.

Reply:
code please :) not sure what you mean :p

Reply:
https://github.com/ExtendRealityLtd/Zinnia.Unity/blob/0649c35f7d920424d4caa59abc10df3f9b2b3092/Runtime/Tracking/Collision/CollisionNotifier.cs#L174-L177
https://github.com/ExtendRealityLtd/Zinnia.Unity/blob/0649c35f7d920424d4caa59abc10df3f9b2b3092/Runtime/Tracking/Collision/CollisionNotifier.cs#L207